### ISSUE TITLE
Support websocket connections to endpoints which return 301/302 redirects

### ISSUE THREAD

**Issue by tompntn** [crossbario/autobahn-python#1488]
Hi,

At present, if a websocket connection is attempted to a URL which returns a 301 or a 302 HTTP code, then Autobahn will throw an error: `WebSocket connection upgrade failed (301 - MovedPermanently)`.

Looking in the code, I can see a "FIXME: support redirects" (https://github.com/crossbario/autobahn-python/blob/da419362239fa0a758d6c7d9c8c3f6d3aa910cd2/autobahn/websocket/protocol.py#L3676-L3691).

It would be awesome if redirects were supported.
Thank you!

**Comment by Munikumar09** [comment]
I am also facing the same problem. Is there any update or solution for this?

**Comment by oberstet** [comment]
yes, fwiw, just to add some pointers: while Autobahn support HTTP redirects as a WebSocket _Server_

https://github.com/crossbario/autobahn-python/blob/7bc85b34e200640ab98a41cfddb38267f39bc92e/autobahn/websocket/protocol.py#L3158
https://github.com/crossbario/autobahn-python/blob/7bc85b34e200640ab98a41cfddb38267f39bc92e/autobahn/websocket/protocol.py#L2711

it doesn't follow HTTP redirects as a WebSocket _Client_

https://github.com/crossbario/autobahn-python/blob/7bc85b34e200640ab98a41cfddb38267f39bc92e/autobahn/websocket/protocol.py#L3556

----

which of **HTTP 30X Status Codes**

- https://developer.mozilla.org/de/docs/Web/HTTP/Status/301
- https://developer.mozilla.org/de/docs/Web/HTTP/Status/303
- https://developer.mozilla.org/de/docs/Web/HTTP/Status/307
- https://developer.mozilla.org/de/docs/Web/HTTP/Status/308
- others?

make sense / should be supported is a related side/detail question.

----

implementing this is straight forward .. I think, if I haven't overlooked sth .. and could be done without big risks (regressions) as the code is pretty confined, and in any case only involves the initial HTTP opening handshake of WebSocket
